cmd/createorupdate/createorupdate.go

```diff
@@ -252,7 +253,7 @@ func main() {
 	adminClient.Authorizer = authorizer
 	v20180930previewClient.Authorizer = authorizer
 
-	ctx, cancel := context.WithTimeout(context.Background(), *timeout)
+	ctx, cancel := fakerp.GetContext(1 * time.Hour)
```

Comment:
I'd like to see this being `ctx := fakerp.EnrichContext(context.Background())` and the timeout flag to die in a fire :)

Reply:
The problem is that you have different types of requests coming from the client (create takes ~15-20m, update is variable based on what is updated, restore is ~30m, key rotation is the longest since everything needs to get updated (~40-50m).

Reply:
deletes are ~10-15m

Reply:
we need timeout, as if somebody messes the PR, it might block CI for quite a while :/ 
Added it configurable via flag

Reply:
why? we already have code that uses /usr/bin/timeout, why do we need to maintain our own implementation of that inside createorupdate too?